docs/hard-fork-upgrade/prerequisites-and-planning.mdx

```diff
@@ -0,0 +1,154 @@
+---
+title: Archive migration prerequisites
+sidebar_label: Archive migration prerequisites
+hide_title: true
+description: Mina protocol havily relies on its archive node. It is crucial to properly migrate archives from Mainnet to Berkeley.
+keywords:
+  - hard fork
+  - upgrade
+  - archive migration
+  - planning 
+  - prerequisites
+  - mina archive node
+  - archive node
+---
+
+# Migrating Mainnet Database to Berkeley
+
+Before you start planning the migration of your Mainnet database into the Berkeley version of the Mina network, read this document in entirety.
+
+## Overview
+
+Mainnet database requires to be migrated in two steps. Each step uses an application that is shipped by Mina Foundation and o1Labs teams:
```

Comment:
Mainnet database is migrated in two steps. ???

Reply:
tbh, I might have introduced this error
@dkijania  can you make sure my suggestions to change to archive node database are valid? thankies

Reply:
@halsaphi, Yes. Please refer to RFC for data migration: https://github.com/MinaProtocol/mina/blob/develop/rfcs/0056-hard-fork-data-migration.md#detailed-design 

In short:

Basically, the berkeley_migration tool does the migration from old db to the new one. But there're some tables can not be populated just by querying the mainnet db. That's why we need a second app to replay all the blocks from the genesis.